automation/src/main/java/org/wso2/carbon/testgrid/automation/executers/TestNgExecuter.java

```diff
@@ -18,23 +18,65 @@
 
 package org.wso2.carbon.testgrid.automation.executers;
 
+import org.testng.TestNG;
 import org.wso2.carbon.testgrid.automation.exceptions.TestGridExecuteException;
 import org.wso2.carbon.testgrid.automation.executers.common.TestExecuter;
 import org.wso2.carbon.testgrid.common.Deployment;
 
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+
 /**
  * This class is responsible for Executing TestNG tests.
  */
 public class TestNgExecuter implements TestExecuter {
+    private String testGridFolder;
+    private String testName;
 
     @Override
-    public void execute(String script, Deployment deployment) {
+    public void execute(String jarFilePath, Deployment deployment) {
+        File jarFile = new File(jarFilePath);
```

Comment:
Please use the NIO File API instead.

Reply:
This is used to create a file instance. NIO File API is used for file operations and not for creation.